clippy_lints/src/swap.rs

```diff
@@ -113,6 +113,18 @@ fn generate_swap_warning(cx: &LateContext<'_>, e1: &Expr<'_>, e2: &Expr<'_>, spa
 
     let first = Sugg::hir_with_applicability(cx, e1, "..", &mut applicability);
     let second = Sugg::hir_with_applicability(cx, e2, "..", &mut applicability);
+    let sugg = if !is_no_std_crate(cx) {
+        "std"
+    } else if is_no_core_crate(cx) {
+        ""
+    } else {
+        "core"
+    };
```

Comment:
Could use one more util `std_or_core() -> Option<&'static str>`. Shouldn't use empty strings as "none"`.

Reply:
Was going for a direct return there on the last else statement, but I did this as a workaround.

```rust
    let sugg = if !is_no_std_crate(cx) {
        "std"
    } else if !is_no_core_crate(cx) {
        "core"
    } else {
        return;
    };
```

Now that https://github.com/rust-lang/rust-clippy/pull/7895 is merged, I think we can go for this, wdyt?

Reply:
I think `std_or_core` would be good to have in `clippy_utils`. You could sneak in a let-else.

```rust
let Some(sugg) = std_or_core(cx) else { return };
```

Reply:
`let-else` sounds amazing

Reply:
Oh whoops, `semicolon-if-nothing-returned` FP on this?

```rust
let Some(sugg) = std_or_core(cx) else { return };
```

produces

```
   --> src/swap.rs:116:5
    |
116 |     let Some(sugg) = std_or_core(cx) else { return };
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: add a `;` here: `let Some(sugg) = std_or_core(cx) else { return };;`
    |
    = note: `-D clippy::semicolon-if-nothing-returned` implied by `-D clippy::pedantic`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#semicolon_if_nothing_returned
```

I will open an issue for this 😅